src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,8 +1,47 @@
 package core.basesyntax;
 
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
 
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder monthReport = new StringBuilder();
+        monthReport
+                .append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append("\n");
+
+        for (String name : names) {
+            int salary = 0;
+            for (String dataItem : data) {
+                String[] dataParts = dataItem.split(" ");
+
+                try {
+                    Date from = new SimpleDateFormat("dd.MM.yyyy").parse(dateFrom);
```

Comment:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#if-you-create-a-formatter-make-it-a-constant-field